_posts/2019-08-13-how-to-deploy-production-grade-vpc-aws.adoc

```diff
@@ -683,25 +684,29 @@ provider "aws" {
 }
 
 terraform {
+  # TODO: this didn't work for me. after creating the region/account/common/terragrunt hcl files, it complained that
+  # backend was defined more than once.
```

Comment:
This is actually caused by a difference in the approach to backend state management. If you follow the method described in https://gruntwork.io/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library/#deploy_using_terragrunt, you will actually need this backend statement since it is not using the generator pattern.

That said, we should definitely update the "Deploy using Terragrunt" guide to modernize the terragrunt approach to use the generator pattern, and then update all the other guides to assume that approach.

Reply:
Ah! Interesting about the two patterns. Thank you for highlighting that.

> That said, we should definitely update the "Deploy using Terragrunt" guide to modernize the terragrunt approach to use the generator pattern, and then update all the other guides to assume that approach.

Roger!

- [ ] update Deploy using Terragrunt to use generator pattern
- [ ] Update all guides to assume that approach

Reply:
For reference, here is the docs on the generator pattern (https://terragrunt.gruntwork.io/docs/features/keep-your-remote-state-configuration-dry/#using-the-generate-property-to-generate-terraform-code-for-managing-remote-state)

Reply:
Okay, made my best guess here to update the other guide.